### PR TITLE
DOCSP-41142: Work with Indexes + Single Field Indexes

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41142>
Staging - 
* <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41142-single-field-indexes/work-with-indexes/>
* <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41142-single-field-indexes/indexes/single-field-index/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes/single-field-index.txt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+.. literalinclude:: /includes/indexes/indexes.kt
+   :start-after: start-movie-class
+   :end-before: end-movie-class
+   :language: kotlin
+   :copyable:
+
+Create Single-Field Index
+-------------------------
+
+The following example creates an index in ascending order on the ``title`` field:
```
Comment (shuangela) [inline source/indexes/single-field-index.txt:52]: Q:Are there options to create an index in any other order other than ascending and descending?

S: Clarify/state explicitly that you can create an index in either ascending or descending order.

File: source/indexes/single-field-index.txt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      :end-before: end-index-single-query
+      :language: kotlin
+      :dedent:
+   
+   .. output::
+      :visible: false
+
+      Movie(id=573a1398f29313caabceb515, title=Batman, ...)
+
+To learn more, see :manual:`Single Field Indexes </core/index-single>` in the {+mdb-server+} manual.
```
Comment (shuangela) [inline source/indexes/single-field-index.txt:77]: S: add api documentation at the end of this page

**Review by shuangela — CHANGES_REQUESTED** [review]
nice work, some comments/questions for you.

File: source/work-with-indexes.txt [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+.. literalinclude:: /includes/indexes/indexes.kt
+   :language: kotlin
+   :start-after: start-remove-index
+   :end-before: end-remove-index
+   :dedent:
+
+.. note::
+
+   You cannot remove a single field from a compound text index. You must
```
[the remark this answers is not in the document]
Comment (mcmorisi) [reply source/work-with-indexes.txt:101]: All types of indices are slated to receive their own page – this page will link to the compound text index page once it's written.

File: source/work-with-indexes.txt [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+To improve query performance, build indexes on fields that appear often in
+your application's queries and operations that return sorted results. Each
+index that you add consumes disk space and memory when active, so we recommend
+that you track index memory and disk usage for capacity planning. In addition,
+when a write operation updates an indexed field, MongoDB updates the related
+index.
+
+Because MongoDB supports dynamic schemas, applications can query against fields
+whose names are not known in advance or are arbitrary. MongoDB 4.2 introduced
```
[the remark this answers is not in the document]
Comment (mcmorisi) [reply source/work-with-indexes.txt:47]: Can reword.

File: source/work-with-indexes.txt [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+.. note::
+
+   You cannot remove a single field from a compound text index. You must
+   drop the entire index and create a new one to update the indexed
+   fields.
+
+Delete All Indexes
+~~~~~~~~~~~~~~~~~~
+
+Starting with MongoDB 4.2, you can drop all indexes by calling the
```
[the remark this answers is not in the document]
Comment (mcmorisi) [reply source/work-with-indexes.txt:108]: Can remove

File: source/indexes/single-field-index.txt (hunk 1 above)
Comment (mcmorisi) [reply source/indexes/single-field-index.txt:52]: Will add to page overview.

**Review by shuangela — APPROVED** [review]
lgtm 💯

**Pull request merged**